packages/relay/src/lib/db/repositories/hbarLimiter/ethAddressHbarSpendingPlanRepository.ts

```diff
@@ -68,11 +68,16 @@ export class EthAddressHbarSpendingPlanRepository {
    *
    * @param {IEthAddressHbarSpendingPlan} addressPlan - The plan to save.
    * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   * @param {number} [ttl] - The time-to-live for the cache entry. (default: 1 day)
    * @returns {Promise<void>} - A promise that resolves when the ETH address is linked to the plan.
    */
-  async save(addressPlan: IEthAddressHbarSpendingPlan, requestDetails: RequestDetails): Promise<void> {
+  async save(
+    addressPlan: IEthAddressHbarSpendingPlan,
+    requestDetails: RequestDetails,
+    ttl: number = this.oneDayInMillis,
```

Comment:
Hmm, might be better to set this to be `this.limitDuration` by default, because if the `limitDuration` is more than one day, we would lose the BASIC spending plans before the hbar limiter has been restarted. Good catch!

Reply:
Changed!